src/software/new_geom/point.h

```diff
@@ -0,0 +1,451 @@
+#pragma once
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/polygon/point_concept.hpp>
+#include <cmath>
+#include <iostream>
+
+#include "software/new_geom/angle.h"
+#include "software/new_geom/vector.h"
+
+/**
+ * A point in 2D space.
+ */
+class Point final
+{
+   public:
+    /**
+     * Creates a Point at the origin (0, 0).
+     */
+    explicit constexpr Point();
+
+    /**
+     * Creates a unit-magnitude Point from an angle.
+     *
+     * @param angle the angle
+     *
+     * @return Point the Point
+     */
+    static Point createFromAngle(const Angle &angle);
```

Comment:
Does it make sense for Points to be created from Angle? I would think this is more appropriate for Vector.